tensorflow_addons/utils/test_utils.py

```diff
@@ -16,15 +16,21 @@
 
 import os
 import random
+import threading
 
 import numpy as np
 import pytest
 import tensorflow as tf
+from tensorflow import keras
 
 from tensorflow_addons.utils import resource_loader
-
-# TODO: copy the layer_test implementation in Addons.
-from tensorflow.python.keras.testing_utils import layer_test  # noqa: F401
+from tensorflow.python.framework import (
+    test_util,
+    tensor_shape,
+    tensor_spec,
+)  # noqa: F401
+from tensorflow.python.util import tf_inspect  # noqa: F401
+from tensorflow.python.eager import context  # noqa: F401
```

Comment:
We would like to remove all private API from TF. Like `tensor_spec.TensorSpec`, we can directly use `tf.TensorSpec` instead; or we can use `tf.executing_eagerly()` rather than `context.executing_eagerly()`. Could we find all alternatives? Thank you!

https://www.tensorflow.org/api_docs/python/tf/TensorSpec